src/llamafactory/chat/hf_engine.py

```diff
@@ -163,10 +184,26 @@ def _process_args(
             generation_config=GenerationConfig(**generating_args),
             logits_processor=get_logits_processor(),
         )
+        if image is not None and model_args.visual_inputs_type == "glm4v_like":
+            transform = torchvision.transforms.Compose(
+                [
+                    torchvision.transforms.Resize(
+                        (1120, 1120), interpolation=torchvision.transforms.InterpolationMode.BICUBIC
+                    ),
+                    torchvision.transforms.ToTensor(),
+                    torchvision.transforms.Normalize((0.48145466, 0.4578275, 0.40821073), (0.26862954, 0.26130258, 0.27577711)),
+                ]
+            )
+            gen_kwargs["images"] = transform(Image.fromarray(image)).unsqueeze(0).to(model.device).to(model_args.compute_dtype)
+        elif model_args.visual_inputs_type == "glm4v_like":
+            gen_kwargs["images"] = None
 
         if pixel_values is not None:
             gen_kwargs["pixel_values"] = pixel_values
 
+        if image_sizes is not None and model_args.visual_inputs_type == "phi3v_like":
+            gen_kwargs["image_sizes"] = image_sizes
```

Comment:
In my pr https://github.com/hiyouga/LLaMA-Factory/pull/4136, I add `image_data_key` and `video_data_key` to template to apply for more models.

You can just keep this for now, and modify in the future.